lib/monitor/measure_test.go

```diff
@@ -0,0 +1,57 @@
+package monitor
+
+import (
+	"testing"
+	"time"
+)
+
+type DummyCounterIO struct {
+	rvalue uint64
+	wvalue uint64
+}
+
+func (dm *DummyCounterIO) Rx() uint64 {
+	dm.rvalue += 10
+	return dm.rvalue
+}
+
+func (dm *DummyCounterIO) Tx() uint64 {
+	dm.wvalue += 10
+	return dm.wvalue
+}
+
+func TestCounterIOMeasureRecord(t *testing.T) {
+	setupMonitor(t)
+	dm := &DummyCounterIO{0, 0}
+	// create the counter measure
+	cm := NewCounterIOMeasure("dummy", dm)
+	if cm.baseRx != dm.rvalue || cm.baseTx != dm.wvalue {
+		t.Logf("baseRx = %d vs rvalue = %d || baseTx = %d vs wvalue = %d", cm.baseRx, dm.rvalue, cm.baseTx, dm.wvalue)
+		t.Fatal("Tx() / Rx() not working ?")
+	}
+	//bread, bwritten := cm.baseRx, cm.baseTx
+	cm.Record()
+	// check the values again
+	if cm.baseRx != dm.rvalue || cm.baseTx != dm.wvalue {
+		t.Fatal("Record() not working for CounterIOMeasure")
+	}
+
+	// TODO do a normal test with single measure passing by the Monitor to the
```

Comment:
OK, that's not good. So the test should be back in, but in fact it won't work, as I saw in #263. This is because the values `.Avg()` are not updated before you call `Collect` or so.